tests/foreman/ui/test_ansible.py

```diff
@@ -275,6 +275,71 @@ def test_positive_host_role_information(target_sat, function_host):
         assert all_assigned_roles_table[0]["Name"] == SELECTED_ROLE
 
 
+@pytest.mark.rhel_ver_match('8')
+def test_positive_verify_ansible_variable_on_host(
+    target_sat, rhel_contenthost, module_activation_key, module_org, module_location
+):
+    """Verify ansible variable is added to the role and attached to the host.
+
+    :id: 7ec4fe19-5a08-4b10-bb4e-7327dd68699a
+
+    :BZ: 2170727
+
+    :customerscenario: true
+
+    :steps:
+
+        1. Create an Ansible variable with array type and set the default value.
+        2. Enable both 'Merge Overrides' and 'Merge Default'.
+        3. Add the variable to a role and attach the role to the host.
+        4. Verify that ansible variable is added to the host.
+
+    :expectedresults: The variable is successfully added to the host.
+    """
+    key = gen_string('alpha')
+    SELECTED_ROLE = 'redhat.satellite.activation_keys'
+    proxy_id = target_sat.nailgun_smart_proxy.id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': proxy_id, 'role_names': [SELECTED_ROLE]})
+    command = target_sat.api.RegistrationCommand(
+        organization=module_org,
+        location=module_location,
+        activation_keys=[module_activation_key.name],
+    ).create()
+    result = rhel_contenthost.execute(command)
```

Comment:
As you're not using the host here, so you could skip the rhel_contenthost and its registration, instead create a fake host without registration and assign the role directly

Reply:
Good to use the actual host here.

Reply:
Either way, this should be setup.

Reply:
yes, it is part of setup, but we're using fake hosts everywhere for such tests for Ansible component, also this will make the execution faster and will save time for unnecessary checkout of actual host

Reply:
It is using container so it doesn't take much time.